src/NUnitEngine/nunit.engine.core/Services/DriverService.cs

```diff
@@ -127,5 +118,24 @@ public override void StartService()
                 throw;
             }
         }
+
+        private void InitializeDriverFactories()
+        {
+            _factories = new List<IDriverFactory>();
+
+            if (_extensionService != null)
+            {
+                foreach (IDriverFactory factory in _extensionService.GetExtensions<IDriverFactory>())
+                    _factories.Add(factory);
```

Comment:
This could be switched to use `_factories.AddRange(...)` instead of manually iterating.